services/121-service/src/registration/repositories/registration-scoped.repository.ts

```diff
@@ -111,9 +112,22 @@ export class RegistrationScopedRepository extends RegistrationScopedBaseReposito
     return this.repository.update(criteria, partialEntity);
   }
 
-  async getRegistrationByReferenceId(referenceId: string) {
+  public async getRegistrationByReferenceId({
```

Comment:
Can the signature of this function be simpler? Without optional params and without a "relations" setting?

Reply:
Well, yes & no. This is a side effect of this method being used in several places/contexts. So to simplify this specific signature we would need define more methods/functions with the needed signatures.

Reply:
Yes, indeed. IMO that would be cleaner (i.e. simpler to understand and less likely to introduce a bug/security risk).
So something like various functions with names like:
getRegistrationAnd...(relation name)
getRegistrationByReferenceIdAndProgramId

The current method can become private and used by these various public methods, to avoid code duplication.